### ISSUE TITLE
Crash on close if used --no-video-playback option

### ISSUE THREAD

**Issue by Bonoboo** [Genymobile/scrcpy#4107]
- [x] I have read the [FAQ](https://github.com/Genymobile/scrcpy/blob/master/FAQ.md).
 - [x] I have searched in existing [issues](https://github.com/Genymobile/scrcpy/issues).

**Environment**
 - OS: Windows 7 x64 (yup, outdated)
 - scrcpy version: 2.1
 - installation method: Windows release
 - device model: Redmi Note 10 Pro
 - Android version: 11

**Describe the bug**
If just start `scrcpy.exe` - all fine. If add only `--no-video-playback` all works too until closing "console" window or Scrcpy's window if it launched directly. Then app crashes with default Windows message "scrcpy.exe stopped working". This makes impossible using command `--kill-adb-on-close`

Here info about crash from EventViewer, it's always the same:
```
  <System>
    <Provider Name="Application Error" />
    <EventID Qualifiers="0">1000</EventID>
    <Level>2</Level>
    <Task>100</Task>
    <Keywords>0x80000000000000</Keywords>
    <TimeCreated SystemTime="2023-06-23T09:14:04.000000000Z" />
    <EventRecordID>1922268</EventRecordID>
    <Channel>Application</Channel>
    <Computer>PC</Computer>
    <Security />
  </System>
  <EventData>
    <Data>scrcpy.exe</Data>
    <Data>0.0.0.0</Data>
    <Data>649384f3</Data>
    <Data>ole32.dll</Data>
    <Data>6.1.7601.26316</Data>
    <Data>639aa8b8</Data>
    <Data>c0000005</Data>
    <Data>000000000001f6ac</Data>
    <Data>5d06c</Data>
    <Data>01d9a5b30ad127fa</Data>
    <Data>C:\Program Files\Android\scrcpy\scrcpy.exe</Data>
    <Data>C:\Windows\system32\ole32.dll</Data>
    <Data>4be56dda-11a6-11ee-beec-705a0f665151</Data>
  </EventData>
</Event>
```
BTW again **_thanks_** for so useful application.

**Comment by rom1v** [comment]
Thank you for your report.

I need more details to reproduce (I fail to reproduce for now).

> If add only `--no-video-playback` all works too until closing "console" window **or Scrcpy's window if it launched directly**.

If you pass `--no-video-playback`, then there is no scrcpy window, right?

To stop, you have to Ctrl+C in the console. Does it also crash in that case?

If you close the console, this kills scrcpy violently (but I don't understand how killing the process would cause a segfault or something which triggers _scrcpy.exe stopped working_).

Is `scrcpy --no-video-playback` your full command line, or do you pass other arguments?

What if you pass `scrcpy --video` instead?

**Comment by Bonoboo** [comment]
> If you pass --no-video-playback, then there is no scrcpy window, right?

Right, no window. I meant window if started from BAT like `start "" scrcpy.exe`
> To stop, you have to Ctrl+C in the console. Does it also crash in that case?

No. And also in this case ADB server killed successfully if add that option.
> Is scrcpy --no-video-playback your full command line, or do you pass other arguments?

Yes. This happens when there is only this single argument.

> What if you pass scrcpy --video instead?

`ambiguous option -- video`

**Comment by rom1v** [comment]
> What if you pass `scrcpy --video` instead?

Sorry, I meant `scrcpy --no-video`.

**Comment by Bonoboo** [comment]
> I meant scrcpy --no-video

It crashes.
> Or even scrcpy without arguments

It crashes too, but only if I close console window. If close window where screen casting - no crash.
Hmm, seems problem not related to argument.

**Comment by rom1v** [comment]
> UPD: tested on 2.0, no crashes.

Interesting. On my Windows test laptop, even 2.1 does not crash.

Simple test: could you test with scrcpy 2.1, but replace SDL2.dll to use the one from scrcpy 2.0. Does it crash?

Advanced test: if you are able to fully [build](https://github.com/Genymobile/scrcpy/blob/master/doc/build.md) the program, it would be interesting to [git bisect](https://git-scm.com/docs/git-bisect) to find from which commit it fails for you.

**Comment by Bonoboo** [comment]
> replace SDL2.dll to use the one from scrcpy 2.0

Still crashes.
> if you are able to fully build the program

I can't, sorry.
Anyway, I definitely can live with that, just pointed out :)
